core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -278,6 +278,14 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     if (pyFiles != null && !isPython) {
       SparkSubmit.printErrorAndExit("--py-files given but primary resource is not a Python script")
     }
+    if (totalExecutorCores != null && executorCores != null) {
```

Comment:
Think about this again, I think this part of code logic could be moved to `SparkConf`, in case user sets these configuration via `SparkConf` object in a programmatic way.